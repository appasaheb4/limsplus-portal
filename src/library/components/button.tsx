import React from "react";
import { IconProps } from "./icon";

interface ButtonProps {
  type?: "solid" | "outline";
  onClick?: () => void;
  size?: "small" | "medium" | "large";
  icon?: React.FunctionComponent<IconProps>;
  pill?: boolean;
  disabled?: string;
}

const Button: React.FunctionComponent<ButtonProps> = (props) => {
  const buttonSizeClass =
    props.size === "small"
      ? `px-2 py-1 text-xs`
      : props.size === "large"
      ? `px-4 py-2 text-base`
      : "px-3 py-2 text-sm";

  const buttonColorClass =
    props.type === "solid"
      ? "text-white bg-indigo-600 hover:bg-indigo-700 hover:shadow-lg"
      : "text-gray-600 border border-gray-400 hover:shadow-lg";

  const roundedClass = props.pill ? "rounded-full" : "rounded-lg";

  const Icon = props.icon;

  return (
    <button
      onClick={props.onClick}
      type="button"
      disabled={!!props.disabled}
      title={props.disabled}
      className={`inline-flex items-center ${buttonSizeClass} ${roundedClass} shadow-sm  font-medium ${buttonColorClass} disabled:opacity-50 disabled:cursor-not-allowed text-center`}
    >
      {Icon && (
        <Icon
          size={props.size}
          type={props.type === "solid" ? "inverse" : "solid"}
          buttonOffset
        />
      )}
      {props.children}
    </button>
  );
};

export default Button;
