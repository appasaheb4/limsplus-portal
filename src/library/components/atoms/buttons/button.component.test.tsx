import React from 'react';
import {render} from '@testing-library/react';
import {Button} from './button.component';

describe('Button component', () => {
  it('render button correctly', () => {
    const button = render(
      <Button className='w-40' onClick={() => jest.fn()}>
        Permanent Address
      </Button>,
    );
    expect(button).toMatchSnapshot();
  });
});
