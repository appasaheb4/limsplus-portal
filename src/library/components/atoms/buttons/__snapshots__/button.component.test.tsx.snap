// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render button correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-40"
      >
        <button
          class="undefined inline-flex items-center px-3 py-2 text-sm rounded-lg shadow-sm   font-medium text-gray-600 border border-gray-400 hover:shadow-lg disabled:opacity-50 disabled:cursor-not-allowed text-center"
          type="button"
        >
          Permanent Address
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-40"
    >
      <button
        class="undefined inline-flex items-center px-3 py-2 text-sm rounded-lg shadow-sm   font-medium text-gray-600 border border-gray-400 hover:shadow-lg disabled:opacity-50 disabled:cursor-not-allowed text-center"
        type="button"
      >
        Permanent Address
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
